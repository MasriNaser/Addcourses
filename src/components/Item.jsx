import React, { Component } from 'react';
import { List, Button, Header } from 'semantic-ui-react';
import PropTypes from 'prop-types';
export default class Item extends Component {
  render() {
    const { name, onDelete, onEdit } = this.props;
    return (
      <List divided verticalAlign='middle'>
        <List.Item>
          <List.Content floated='right'>
            <Button onClick={onDelete}>Delete</Button>
          </List.Content>
          <List.Content floated='right'>
            <Button onClick={onEdit}>Edit</Button>
          </List.Content>
          <List.Content>
            <Header>{name}</Header>
          </List.Content>
        </List.Item>
      </List>
    );
  }
}
Item.prototypes = {
  name: PropTypes.string.isRequired,
  onDelete: PropTypes.func.isRequired,
  onEdit: PropTypes.func.isRequired
};
