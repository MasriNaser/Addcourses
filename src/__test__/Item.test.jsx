import React from 'react';
import { shallow } from 'enzyme';
import Item from '../components/Item';
import toJson from 'enzyme-to-json';

const onDelete = jest.fn().mockName('onDelete');
const onEdit = jest.fn().mockName('onEdit');
const wrapper = shallow(<Item updateForm={onDelete} name='' onEdit={onEdit} />);
describe('Item component', () => {
  it('should take snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should has Form', () => {
    expect(wrapper.find('List').exists()).toBe(true);
  });
  it('should has Button', () => {
    expect(wrapper.find('Button').exists()).toBe(true);
  });
  it('should has Liked component',()=>{
    expect(wrapper.find('Liked').exists()).toBe(true)
  })
});
