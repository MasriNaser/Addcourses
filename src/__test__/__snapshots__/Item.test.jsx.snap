// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component should take snapshot 1`] = `
<List
  divided={true}
  verticalAlign="middle"
>
  <ListItem>
    <ListContent
      floated="right"
    >
      <Button
        as="button"
      >
        Delete
      </Button>
    </ListContent>
    <ListContent
      floated="right"
    >
      <Button
        as="button"
        onClick={[MockFunction onEdit]}
      >
        Edit
      </Button>
    </ListContent>
    <Liked />
    <ListContent>
      <Header />
    </ListContent>
  </ListItem>
</List>
`;
